Source/WebCore/platform/graphics/WidthIterator.cpp

```diff
@@ -196,21 +199,10 @@ void WidthIterator::commitCurrentFontRange(GlyphBuffer& glyphBuffer, unsigned& l
     applyInitialAdvance(glyphBuffer, applyFontTransformsResult.initialAdvance, lastGlyphCount);
     m_currentCharacterIndex = currentCharacterIndex;
 
-    if (widthOfCurrentFontRange && m_fallbackFonts && font != &primaryFont) {
-        // FIXME: This does a little extra work that could be avoided if
-        // glyphDataForCharacter() returned whether it chose to use a small caps font.
-        if (!m_font.isSmallCaps() || character == u_toupper(character))
-            m_fallbackFonts->add(font);
-        else {
-            auto glyphFont = m_font.glyphDataForCharacter(u_toupper(character), m_run.rtl()).font;
-            if (glyphFont != &primaryFont)
-                m_fallbackFonts->add(glyphFont);
-        }
-    }
+    if (widthOfCurrentFontRange && m_fallbackFonts && font != &primaryFont)
+        m_fallbackFonts->add(font);
 
     lastGlyphCount = glyphBuffer.size();
-    font = &newFont;
```

Comment:
I think it's a shame that this line had to get deleted. I kind of feel that committing the current range should automatically set up as much state as possible, so the caller doesn't have to remember to do it themself.